public/js/notifications.js

```diff
@@ -0,0 +1,433 @@
+                        // this._logger.debug(this._prefix + `got message from event-stream: ${event.data}`);
+                    });
+                    this.eventSource.addEventListener('error', () => {
+                        // this._logger.debug(this._prefix +`got an error from event-stream: ${event}`);
+                    });
+                    this.eventSource.addEventListener('icinga2.notification', (event) => {
+                        if (this.hasNotificationPermission() && this.hasNotificationsEnabled()) {
+                            // send to service_worker if the permissions are given and the notifications enabled
+                            this.getServiceWorker()
+                                .then((serviceWorker) => {
+                                    if (serviceWorker) {
+                                        serviceWorker.postMessage(
+                                            JSON.stringify({
+                                                command: 'notification',
+                                                notification: JSON.parse(event.data),
+                                                baseUrl: icinga.config.baseUrl
+                                            })
+                                        );
+                                    }
+                                });
+                        }
+                    });
+                } catch (error) {
+                    // this._logger.error(this._prefix + `got an error: ${error}`);
+                }
+            } else {
+                this.logger.warn(this.prefix + "denied opening event-stream as the notification permissions" +
+                    " are missing or the notifications themselves disabled");
+            }
+        }
+
+        renderHandler (event) {
+            if (event.type === 'rendered') {
+                const _this = event.data.self;
+                let url = new URL(event.delegateTarget.URL);
+
+                /**
+                 * TODO(nc): We abuse the fact that the renderHandler method only triggers when the container
+                 *  in col1 (#main > #col1.container) gets rendered. This can only happen on the main interface for
+                 *  now (might break things if columns are introduced elsewhere in the future).
+                 *  This in turn requires a user to be logged in and their session validated.
+                 *  In the future, we should introduce a proper login event and tie the initial event-stream connection
+                 *  to this specific event (SSO should ALSO trigger the login event as the user lands in the
+                 *  interface with an authenticated session).
+                 */
+                if (_this.initialized === false) {
+                    _this.initialized = true;
+                }
+
+                if (url.pathname === _this.icinga.config.baseUrl + '/account') {
```

Comment:
As above, please return earlier if the condition is not met. This prevents indentation.